### ISSUE TITLE
Cannot prove that V =:= Out

### ISSUE THREAD

**Issue by fggarcia** [outworkers/phantom#531]
```scala
update
    .p_where(_.key eqs ?)
    .p_and(_.user eqs ?)
    .p_and(_.period eqs ?)
    .p_modify(_.values setTo ?)
    .ttl(?)
    .consistencyLevel_=(ConsistencyLevel.QUORUM)
    .prepare()

insertPriceCachePS.bind(
      priceData.key,
      priceData.user,
      columnKeyMetadata.date,
      (priceMetadata._1, createPriceMetadataAsString(priceMetadata._2)),
      ttl
    )
```
this example throw error 
```
Error:(351, 51) not enough arguments for method bind: (implicit tp: shapeless.ops.hlist.Tupler.Aux[shapeless.::[scala.collection.immutable.Map[String,String],shapeless.::[String,shapeless.::[String,shapeless.::[java.util.Date,shapeless.::[Long,shapeless.HNil]]]]],(scala.collection.immutable.Map[String,String], String, String, java.util.Date, Long)], implicit ev: =:=[(String, com.despegar.memento.HotelId, java.util.Date, scala.collection.immutable.Map[com.despegar.memento.PriceMetadataKey,String], Int),(scala.collection.immutable.Map[String,String], String, String, java.util.Date, Long)])com.websudos.phantom.builder.query.prepared.ExecutablePreparedQuery.
Unspecified value parameter ev.
    val insertPriceCache = insertPriceCachePS.bind(

Error:(351, 51) Cannot prove that (String, com.memento.UserId, java.util.Date, scala.collection.immutable.Map[String,String], Int) =:= Out.
    val insertPriceCache = insertPriceCachePS.bind(
```
What's wrong with the prepared statement??

**Comment by alexflav23** [comment]
Hi @fggarcia,

Check the types basically. Look at what type the columns have versus the stuff you are binding to. We know the error is not tremendously helpful and we are working to improve it, but in the mean time just check that. It's telling you the tuple is of the wrong type checked.

Hope this helps! And btw the `ttl` is an `Int` not a `Long`, I was mislead by that as well somehow.

Regards,
Flavian

**Comment by fggarcia** [comment]
Hi @alexflav23 sorry for not put the table diagram in the post.

object key extends StringColumn(this) with PartitionKey[String]
  object user extends StringColumn(this) with PrimaryKey[String] with ClusteringOrder[String] with Ascending
  object period extends DateColumn(this) with PrimaryKey[Date] with ClusteringOrder[Date] with Ascending
  object values extends MapColumn[String, String](this)

All of the types are basics, nothing special.

I believe that the problem is in the MapColumn

We try to replace the next code:

    val insertPriceCache = priceCheapestCacheStore.update
      .where(_.user eqs priceData.key)
      .and(_.period eqs columnKeyMetadata.date)
      .modify(_.values put (priceMetadata._1 , createPriceMetadataAsString(priceMetadata._2)))
      .ttl(ttl)
      .consistencyLevel_=(ConsistencyLevel.QUORUM)

**Comment by alexflav23** [comment]
Hi @fggarcia,

It seems you are trying to use  prepared version of `put` which we don't currently have, not 100% sure this is support in Cassandra but certainly can be checked, no reason why it shouldn't. 

Could that be the problem, you are binding a tuple where the statement expects a Map?

**Comment by alexflav23** [comment]
Here's how to fix your particular problem. The Scala compiler isn't particularly helpful at printing the `Out` type properly, that's basically an `HList` that gets build as you call methods that take a `?` as an argument and then reversed when you bind, and transformed to a `TupleN`.

The method `bind` as you can see only takes one argument, which means we are exploiting auto-tupling, so when you think you are calling `bind` with multiple arguments what is really happening is you are calling `bind` with a tuple. The type evidence is trying to prove the type of the tuple you put in the `bind` is simply matching the things you put in a bind.

**TLDR:** In this case, just use a `Map` for the map part in the bind, not a tuple. You are using `setTo` on the `MapColumn` which means you are trying to replace the entire map value, intentionally or otherwise.

```scala
update
    .p_where(_.key eqs ?)
    .p_and(_.user eqs ?)
    .p_and(_.period eqs ?)
    .p_modify(_.values setTo ?)
    .ttl(?)
    .consistencyLevel_=(ConsistencyLevel.QUORUM)
    .prepare()

insertPriceCachePS.bind(
      priceData.key,
      priceData.user,
      columnKeyMetadata.date,
      Map(priceMetadata._1, createPriceMetadataAsString(priceMetadata._2)),
      ttl
    )
```

**Comment by fggarcia** [comment]
@alexflav23 can you confirm that Put method isn't supported for prepared statement??

Thanks for data about setTo method
